core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -33,23 +38,50 @@ public TransactionSerializer(final ArtifactSigner signer, final long chainId) {
   }
 
   public String serialize(final Transaction transaction) {
-    final byte[] bytesToSign = transaction.rlpEncode(chainId);
+    if (transaction.isEip1559()) {
+      return toHexString(serializeEip1559(transaction));
+    } else {
+      return toHexString(serializeFrontier(transaction));
+    }
+  }
+
+  private byte[] serializeFrontier(final Transaction transaction) {
+    final SignatureData preSigningSignatureData =
+        new SignatureData(longToBytes(chainId), new byte[] {}, new byte[] {});
+    byte[] bytesToSign = transaction.rlpEncode(preSigningSignatureData);
+
+    SignatureData signatureData = sign(bytesToSign);
+
+    signatureData = TransactionEncoder.createEip155SignatureData(signatureData, chainId);
+    return transaction.rlpEncode(signatureData);
+  }
+
+  private byte[] serializeEip1559(final Transaction transaction) {
+    byte[] bytesToSign = transaction.rlpEncode(null);
+    bytesToSign = prependEip1559TransactionType(bytesToSign);
 
+    final SignatureData signatureData = sign(bytesToSign);
+
+    byte[] serializedBytes = transaction.rlpEncode(signatureData);
+    return prependEip1559TransactionType(serializedBytes);
+  }
+
+  private static byte[] prependEip1559TransactionType(byte[] bytesToSign) {
+    return ByteBuffer.allocate(bytesToSign.length + 1)
+        .put(TransactionType.EIP1559.getRlpType())
+        .put(bytesToSign)
+        .array();
```

Comment:
nit: may have been nicer to use the Bytes type to append

Reply:
This is pinched straight from web3j
https://github.com/web3j/web3j/blob/master/crypto/src/main/java/org/web3j/crypto/TransactionEncoder.java#L120-L125